src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/payment/ServiceRequestAPIHandler.java

```diff
@@ -57,46 +60,47 @@ private CallbackResponse makePaymentServiceReq(CallbackParams callbackParams) {
-                if (caseData.getHearingFeePBADetails() == null || (caseData.getHearingFeePBADetails() != null
-                    && caseData.getHearingFeePBADetails().getServiceReqReference() == null)) {
-                    serviceRequestReference = paymentsService.createServiceRequest(caseData, authToken)
-                        .getServiceRequestReference();
-
-                    caseData = caseData.toBuilder()
-                        .hearingFeePBADetails(SRPbaDetails.builder()
-                                                  .applicantsPbaAccounts(caseData.getApplicantSolicitor1PbaAccounts())
-                                                  .fee(caseData.getHearingFee())
-                                                  .serviceReqReference(serviceRequestReference).build())
-                        .build();
+            if (isHearingFeeServiceRequest(caseData) || isClaimFeeServiceRequest(caseData)) {
+                log.info("calling payment service request {}", caseData.getCcdCaseReference());
+                String serviceRequestReference;
+                SRPbaDetails.SRPbaDetailsBuilder paymentDetails;
+                serviceRequestReference = paymentsService.createServiceRequest(caseData, authToken)
+                    .getServiceRequestReference();
+                paymentDetails = SRPbaDetails.builder()
+                    .applicantsPbaAccounts(caseData.getApplicantSolicitor1PbaAccounts())
+                    .serviceReqReference(serviceRequestReference);
+
+                if (isHearingFeeServiceRequest(caseData)) {
+                    paymentDetails.fee(caseData.getHearingFee());
+                    caseData = caseData.toBuilder().hearingFeePBADetails(paymentDetails.build()).build();
+                } else {
+                    paymentDetails.fee(caseData.getClaimFee());
+                    caseData = caseData.toBuilder().claimIssuedPBADetails(paymentDetails.build()).build();
                 }
             } else {
-                if (caseData.getClaimIssuedPBADetails() == null || (caseData.getClaimIssuedPBADetails() != null
-                    && caseData.getClaimIssuedPBADetails().getServiceReqReference() == null)) {
-                    serviceRequestReference = paymentsService.createServiceRequest(caseData, authToken)
-                        .getServiceRequestReference();
-
-                    caseData = caseData.toBuilder()
-                        .claimIssuedPBADetails(SRPbaDetails.builder()
-                                                   .applicantsPbaAccounts(caseData.getApplicantSolicitor1PbaAccounts())
-                                                   .fee(caseData.getClaimFee())
-                                                   .serviceReqReference(serviceRequestReference).build())
-                        .build();
-                }
+                log.info("Service Request is already requested for this case {}", caseData.getCcdCaseReference());
```

Comment:
This addition seems gratuitous. I don't think it's necessary. Let's take it out.

Also, be mindful of using the "info" log level. It should be applied only when the information is important in some way as normally logs are on "info" in an attempt of keeping them small.
This is more for debugging than anything else so should be at debug level, as should the other logs here except for the error.